src/test/java/com/puppycrawl/tools/checkstyle/utils/TokenUtilTest.java

```diff
@@ -341,58 +341,140 @@ public void testIsTypeDeclaration() {
     }
 
     @Test
-    public void testIsOfTypeTrue() {
+    public void testIsBooleanLiteralType() {
+        assertWithMessage("Result is not expected")
+                .that(TokenUtil.isBooleanLiteralType(TokenTypes.LITERAL_TRUE))
+                .isTrue();
+        assertWithMessage("Result is not expected")
+                .that(TokenUtil.isBooleanLiteralType(TokenTypes.LITERAL_FALSE))
+                .isTrue();
+        assertWithMessage("Result is not expected")
+                .that(TokenUtil.isBooleanLiteralType(TokenTypes.LOR))
+                .isFalse();
+    }
+
+    @Test
+    public void testAreOnSameLine() {
+        final int type = TokenTypes.COMPILATION_UNIT;
+        final DetailAstImpl astForTest = new DetailAstImpl();
+        final DetailAstImpl astForTest2 = new DetailAstImpl();
+        astForTest.setLineNo(type);
+        astForTest2.setLineNo(type);
+        final boolean result = TokenUtil.areOnSameLine(astForTest, astForTest2);
+
+        assertWithMessage("Token type did not match")
+            .that(result)
+            .isTrue();
+    }
+
+    @Test
+    public void testIsRootNode() {
+        final int type = TokenTypes.COMPILATION_UNIT;
+        final DetailAstImpl astForTest = new DetailAstImpl();
+        astForTest.setType(type);
+        final boolean result = TokenUtil.isRootNode(astForTest);
+
+        assertWithMessage("Token type did not match")
+            .that(result)
+            .isTrue();
+    }
+
+    /**
+     * IsOfType{} method with 2 parameter is a general test and not used
+     * in Check UTs as we can't hit all conditions to achieve 100%
+     * Code Coverage.
+     */
+    @Test
+    public void testIsOfTypeSecond() {
+        final int type = TokenTypes.LITERAL_CATCH;
+        final DetailAstImpl astForTest = new DetailAstImpl();
```

Comment:
please remove this method and all other methods that do "new DetailAstImpl" that you introducing.
All coverage should be done by Input files and whole Checstyle execution and it is done for Checks. 

You need to extend test of Checks that are using such methods.

Reply:
@romani please check my generated report and comment to see my proposal for this issue.

Reply:
@romani We should think of adding https://checkstyle.org/config_coding.html#IllegalInstantiation for this .

Reply:
@romani 
Here: https://github.com/checkstyle/checkstyle/pull/11543#issuecomment-1105316549